From 6b33c9d5615779a32247dcad31eb290c85a01162 Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Thu, 28 Jan 2016 22:58:07 +0700
Subject: [PATCH 2/2] linker: fix spamming

Change-Id: I854d543b42837a25e68a0018e5190bbffd4e275c
---
 linker/linker.cpp | 2 --
 1 file changed, 2 deletions(-)

diff --git a/linker/linker.cpp b/linker/linker.cpp
index eb64939..d9ff917 100644
--- a/linker/linker.cpp
+++ b/linker/linker.cpp
@@ -1988,8 +1988,6 @@ static bool soinfo_link_image(soinfo* si, const android_dlextinfo* extinfo) {
     if (si->has_text_relocations) {
         // Make segments writable to allow text relocations to work properly. We will later call
         // phdr_table_protect_segments() after all of them are applied and all constructors are run.
-        DL_WARN("%s has text relocations. This is wasting memory and prevents "
-                "security hardening. Please fix.", si->name);
         if (phdr_table_unprotect_segments(si->phdr, si->phnum, si->load_bias) < 0) {
             DL_ERR("can't unprotect loadable segments for \"%s\": %s",
                    si->name, strerror(errno));
-- 
2.5.0

