diff --git a/src/java/com/android/internal/telephony/IccSmsInterfaceManager.java b/src/java/com/android/internal/telephony/IccSmsInterfaceManager.java
index 000dce2..60d02a0 100644
--- a/src/java/com/android/internal/telephony/IccSmsInterfaceManager.java
+++ b/src/java/com/android/internal/telephony/IccSmsInterfaceManager.java
@@ -844,7 +844,8 @@ synchronized public boolean enableGsmBroadcastRange(int startMessageId, int endM
                 Binder.getCallingUid());
 
         if (!mCellBroadcastRangeManager.enableRange(startMessageId, endMessageId, client)) {
-            log("Failed to add GSM cell broadcast subscription for MID range " + startMessageId
+            if (DBG)
+                log("Failed to add GSM cell broadcast subscription for MID range " + startMessageId
                     + " to " + endMessageId + " from client " + client);
             return false;
         }
@@ -871,7 +872,8 @@ synchronized public boolean disableGsmBroadcastRange(int startMessageId, int end
                 Binder.getCallingUid());
 
         if (!mCellBroadcastRangeManager.disableRange(startMessageId, endMessageId, client)) {
-            log("Failed to remove GSM cell broadcast subscription for MID range " + startMessageId
+            if (DBG)
+                log("Failed to remove GSM cell broadcast subscription for MID range " + startMessageId
                     + " to " + endMessageId + " from client " + client);
             return false;
         }
@@ -898,7 +900,8 @@ synchronized public boolean enableCdmaBroadcastRange(int startMessageId, int end
                 Binder.getCallingUid());
 
         if (!mCdmaBroadcastRangeManager.enableRange(startMessageId, endMessageId, client)) {
-            log("Failed to add cdma broadcast subscription for MID range " + startMessageId
+            if (DBG)
+                log("Failed to add cdma broadcast subscription for MID range " + startMessageId
                     + " to " + endMessageId + " from client " + client);
             return false;
         }
@@ -925,7 +928,8 @@ synchronized public boolean disableCdmaBroadcastRange(int startMessageId, int en
                 Binder.getCallingUid());
 
         if (!mCdmaBroadcastRangeManager.disableRange(startMessageId, endMessageId, client)) {
-            log("Failed to remove cdma broadcast subscription for MID range " + startMessageId
+            if (DBG)
+                log("Failed to remove cdma broadcast subscription for MID range " + startMessageId
                     + " to " + endMessageId + " from client " + client);
             return false;
         }
