diff --git a/libcutils/android_reboot.c b/libcutils/android_reboot.c
index b3a25fe..a0397a0 100644
--- a/libcutils/android_reboot.c
+++ b/libcutils/android_reboot.c
@@ -97,7 +97,11 @@ static void remount_ro(void)
 
     /* Now poll /proc/mounts till it's done */
+#ifdef STE_HARDWARE
+    while (!remount_ro_done() && (cnt < 50)) {
+#else
     while (!remount_ro_done() && (cnt < 3600)) {
+#endif
         usleep(100000);
         cnt++;
     }
 
