From 55ae6141deda990a6e68b37e840ece74e4b3b554 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=C5=81ukasz=20Domeradzki?= <JustArchi@JustArchi.net>
Date: Sat, 27 Dec 2014 04:47:32 +0100
Subject: [PATCH 7/8] Fix segfault

---
 media/libmediaplayerservice/Android.mk | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/media/libmediaplayerservice/Android.mk b/media/libmediaplayerservice/Android.mk
index 5e02313..362f605 100644
--- a/media/libmediaplayerservice/Android.mk
+++ b/media/libmediaplayerservice/Android.mk
@@ -60,6 +60,8 @@ endif
 
 LOCAL_MODULE:= libmediaplayerservice
 
+LOCAL_CFLAGS += -Os
+
 ifeq ($(TARGET_ENABLE_QC_AV_ENHANCEMENTS),true)
     LOCAL_CFLAGS += -DENABLE_AV_ENHANCEMENTS
     LOCAL_C_INCLUDES += $(TOP)/frameworks/av/include/media
-- 
1.9.1

