From ee9748ff91589bc8d8600db851bdcabd87a1f1b4 Mon Sep 17 00:00:00 2001
From: Chet Kener <Cl3Kener@gmail.com>
Date: Fri, 7 Nov 2014 10:22:52 -0500
Subject: [PATCH 31/95] Disable loop unrolling

Signed-off-by: Chet Kener <Cl3Kener@gmail.com>
---
 media/libstagefright/codecs/Android.mk | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/media/libstagefright/codecs/Android.mk b/media/libstagefright/codecs/Android.mk
index 2e43120..9348969 100644
--- a/media/libstagefright/codecs/Android.mk
+++ b/media/libstagefright/codecs/Android.mk
@@ -2,3 +2,5 @@ LOCAL_PATH:= $(call my-dir)
 include $(CLEAR_VARS)
 
 include $(call all-makefiles-under,$(LOCAL_PATH))
+
+LOCAL_CFLAGS := -fno-unroll-loops
-- 
2.5.0

