From e511d20367bc0f5d2b386f637dac649d7a0832c2 Mon Sep 17 00:00:00 2001
From: Bernhard Rosenkraenzer <Bernhard.Rosenkranzer@linaro.org>
Date: Thu, 25 Jul 2013 12:41:18 +0530
Subject: [PATCH 33/95] downmix: Fix aliasing violation

Signed-off-by: Amit Pundir <amit.pundir@linaro.org>
Signed-off-by: Chet Kener <Cl3Kener@gmail.com>
---
 media/libeffects/downmix/EffectDownmix.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/media/libeffects/downmix/EffectDownmix.c b/media/libeffects/downmix/EffectDownmix.c
index 4a41037..1e0bbf4 100644
--- a/media/libeffects/downmix/EffectDownmix.c
+++ b/media/libeffects/downmix/EffectDownmix.c
@@ -401,7 +401,8 @@ static int Downmix_Command(effect_handle_t self, uint32_t cmdCode, uint32_t cmdS
         memcpy(pReplyData, pCmdData, sizeof(effect_param_t) + sizeof(int32_t));
         ALOGV("Downmix_Command EFFECT_CMD_GET_PARAM param %" PRId32 ", replySize %" PRIu32,
                 *(int32_t *)rep->data, rep->vsize);
-        rep->status = Downmix_getParameter(pDownmixer, *(int32_t *)rep->data, &rep->vsize,
+        int32_t rep_data; memcpy(&rep_data, rep->data, sizeof(int32_t));
+        rep->status = Downmix_getParameter(pDownmixer, rep_data /* *(int32_t *)rep->data */, &rep->vsize,
                 rep->data + sizeof(int32_t));
         *replySize = sizeof(effect_param_t) + sizeof(int32_t) + rep->vsize;
         break;
@@ -414,7 +415,8 @@ static int Downmix_Command(effect_handle_t self, uint32_t cmdCode, uint32_t cmdS
             return -EINVAL;
         }
         effect_param_t *cmd = (effect_param_t *) pCmdData;
-        *(int *)pReplyData = Downmix_setParameter(pDownmixer, *(int32_t *)cmd->data,
+        int32_t cmd_data; memcpy(&cmd_data, cmd->data, sizeof(int32_t));
+        *(int *)pReplyData = Downmix_setParameter(pDownmixer, cmd_data /* *(int32_t *)cmd->data*/,
                 cmd->vsize, cmd->data + sizeof(int32_t));
         break;
 
-- 
2.5.0

