From 320b11041f232645707c608dc56baa92b38b5a10 Mon Sep 17 00:00:00 2001
From: Chet Kener <Cl3Kener@gmail.com>
Date: Sun, 14 Dec 2014 01:01:16 -0500
Subject: [PATCH 48/95] Disable Strict Aliasing on libmediaplayerservice

Signed-off-by: Chet Kener <Cl3Kener@gmail.com>
---
 media/libmediaplayerservice/Android.mk | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/media/libmediaplayerservice/Android.mk b/media/libmediaplayerservice/Android.mk
index 2cf5710..f73766a 100644
--- a/media/libmediaplayerservice/Android.mk
+++ b/media/libmediaplayerservice/Android.mk
@@ -6,6 +6,8 @@ LOCAL_PATH:= $(call my-dir)
 
 include $(CLEAR_VARS)
 
+LOCAL_CPPFLAGS := -fno-strict-aliasing
+
 LOCAL_SRC_FILES:=               \
     ActivityManager.cpp         \
     Crypto.cpp                  \
-- 
2.5.0

